### ISSUE TITLE
Self-supervised training

### ISSUE THREAD

**Issue by guipotje** [georg-bn/rotation-steerers#6]
Hi, thank you for open-sourcing the code, really liked the idea of the paper.

Just wondering if you tried to optimize the projection matrix based on proximity in the descriptor space instead of supervising via softmax matching (for example L2 or cosine similarity). This would allow easy training setup with just a dataset of random images. If you didn't, do you think this approach would work well?

Best,

Guilherme.

**Comment by georg-bn** [comment]
Yes, I have tried that and it usually works quite well! I think L1 distance in the descriptor space worked better than L2 when I was experimenting. In the end we used the matching-based training in the paper just for consistency with setting B and C .

**Comment by guipotje** [comment]
Great to know it works well! Thank you!
